From 4ffff55c97f165ec098a7b4cb770bcf62c55babb Mon Sep 17 00:00:00 2001
From: Vishesh Handa <me@vhanda.in>
Date: Fri, 28 Mar 2014 16:09:17 +0100
Subject: [PATCH 12/48] xattr: the size returned can be <= 0

In that case we do not want to create a QByteArray and try to fill the
byte array.
---
 src/file/lib/baloo_xattr_p.h | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/file/lib/baloo_xattr_p.h b/src/file/lib/baloo_xattr_p.h
index 1174cbc..fefc3d9 100644
--- a/src/file/lib/baloo_xattr_p.h
+++ b/src/file/lib/baloo_xattr_p.h
@@ -53,6 +53,13 @@ inline ssize_t baloo_getxattr(const QString& path, const QString& name, QString*
     const ssize_t size = extattr_get_file(encodedPath, EXTATTR_NAMESPACE_USER, attributeName, NULL, 0);
 #endif
 
+    if (size <= 0) {
+        if (value) {
+            value->clear();
+        }
+        return size;
+    }
+
     QByteArray data(size, Qt::Uninitialized);
 
 #if defined(Q_OS_LINUX)
-- 
1.9.1

